### PR TITLE
Nrf oberon v3.0.5 doc

### PR DESCRIPTION
**Pull request by frkv**
Updates to nrf_oberon (for v3.0.5)
Updates to nrf_security for obeorn backend
Added missing advanced configuration documentation

Ref: NCSDK-5489

### CODE REVIEW

File: nrf_security/doc/nrf_security.rst [hunk 1]
```diff
@@ -81,6 +82,20 @@ To enable the Arm CryptoCell CC310 backend, set the :option:`CONFIG_CC310_BACKEN
...
+
+The :ref:`nrf_oberon_readme` library is a binarly library that provides select cryptographic algorithms optimized for use in nRF SOC's.
+This provides faster execution than standard mbed TLS implementations.
+
+The nrf_oberon backend provides support for AES ciphers, SHA-1 and SHA-256, ECC (ECDH, ECDSA, and ECJPAKE) using secp256r1.
+
+Enabling the nrf_oberon backend
+-------------------------------
+
+To use the :ref:`nrf_oberon_readme` as a backend, set the :option:`CONFIG_OBERON_BACKEND` Kconfig variable to true.
```
Comment (b-gent) [inline nrf_security/doc/nrf_security.rst:96]: this option is not found and cannot be linked when building, is it correct?

File: nrf_security/doc/nrf_security.rst [hunk 2]
```diff
@@ -216,17 +231,22 @@ Multiple backends
...
-| ECB          | Glue           | CC310: :option:`CONFIG_CC310_MBEDTLS_AES_C`     |
-|              |                |                                                 |
-|              |                | Standard: :option:`CONFIG_VANILLA_MBEDTLS_AES_C`|
-+--------------+----------------+-------------------------------------------------+
++--------------+----------------+----------------------------------------------------+
+| Cipher mode  | Support        | Configurations                                     |
++==============+================+====================================================+
+| ECB          | Glue           | CC310: :option:`CONFIG_CC310_MBEDTLS_AES_C`        |
+|              |                |                                                    |
+|              |                | nrf_oberon: :option:`CONFIG_OBERON_MBEDTLS_AES_C`  |
```
Comment (b-gent) [inline nrf_security/doc/nrf_security.rst:239]: this option link doesn't resolve either

**Comment by b-gent** [comment]
@frkv  please check my changes and let me know if all is ok

File: nrf_security/doc/nrf_security.rst (hunk 1 above)
Comment (frkv) [reply nrf_security/doc/nrf_security.rst:96]: This should hopefully been fixed now that we have the actual library in

File: nrf_security/doc/nrf_security.rst (hunk 2 above)
Comment (frkv) [reply nrf_security/doc/nrf_security.rst:239]: This should hopefully been fixed now that we have the actual library in

**Comment by frkv** [comment]
LGTM

**Comment by b-gent** [comment]
Ready to be merged.

**Pull request merged**